handler/src/main/java/io/netty/handler/ssl/JdkAlpnSslEngine.java

```diff
@@ -62,12 +60,15 @@ private static void updateAvailability() {
                     "protocolSelector");
             ALPN.put(engine, new ServerProvider() {
                 @Override
-                public String select(List<String> protocols) {
+                public String select(List<String> protocols) throws SSLException {
                     try {
                         return protocolSelector.select(protocols);
+                    } catch(SSLException e) {
+                        throw e;
                     } catch (Throwable t) {
-                        PlatformDependent.throwException(t);
-                        return null;
+                        // Ensure that all exceptions are propagated as SSLExceptions
+                        // so that the SslHandler properly fails the handshake.
+                        throw new SSLException(t);
```

Comment:
Yes that description for `SSLHandshakeException` is a bit vague.  It is being used by the JDK in stations where no compatible ciphers, protocols, hash algo, signature algo, etc... [see here](http://grepcode.com/file/repository.grepcode.com/java/root/jdk/openjdk/8-b132/sun/security/ssl/Handshaker.java/#491), [here](http://grepcode.com/file/repository.grepcode.com/java/root/jdk/openjdk/8-b132/sun/security/ssl/ServerHandshaker.java?av=f#681), and [here](http://grepcode.com/file/repository.grepcode.com/java/root/jdk/openjdk/8-b132/sun/security/ssl/ClientHandshaker.java?av=f#1010).  It is within the same realm of "we are trying to do a handshake, but didn't find a supported algorithm, so we are failing the handshake".  I tend to lean more toward `SSLHandshakeException` but could also go either way....but think we should be consistent 1 way or the other.